### PR TITLE
openocd make + .deb package creator

### PR DESCRIPTION
**Pull request by barnasm**

### CODE REVIEW

File: toolchain/openocd-mimiker/Makefile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+build: configure
+	cd openocd-build && make -j$(shell nproc --all) LDFLAGS="-Wl,--as-needed"
+	touch $@-stamp
+
+install: build
+	cd openocd-build && make install DESTDIR=$(PWD)/debian/tmp
+	cd $(PWD)/debian/tmp && \
+		mv usr/bin/openocd usr/bin/openocd-mimiker && \
+		mv usr/share/openocd/ usr/share/openocd-mimiker
```
Comment (cahirwpz) [inline toolchain/openocd-mimiker/Makefile]: I don't think it's going to work. Most likely there're some strings embedded into openocd binary that refer to `/usr/share/openocd` directory.

File: toolchain/openocd-mimiker/Makefile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+configure:
+	rm -rf openocd-build
+	mkdir openocd-build
+	cd $(OPENOCD) && \
+	    ./bootstrap
+	cd openocd-build && \
+	    ../$(OPENOCD)/configure \
+	        --prefix=/usr \
+		--enable-jlink 
+	touch $@-stamp
```
Comment (cahirwpz) [inline toolchain/openocd-mimiker/Makefile]: Even though you create stamps this (and others too) rule will trigger again if you run `make` twice.

File: toolchain/openocd-mimiker/Makefile [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		mv usr/share/openocd/ usr/share/openocd-mimiker
+	touch $@-stamp
+
+binary: install
+	fakeroot ./debian/rules binary
+	touch $@-stamp
+
+clean: 
+	dh_clean
+	rm -rf $(OPENOCD) openocd-build
```
Comment (cahirwpz) [inline toolchain/openocd-mimiker/Makefile]: Usually `distclean` rule removes all downloaded files, not `clean`.

**Comment by cahirwpz** [comment]
On freshly cloned repository I get:

```
rm -rf openocd-build
mkdir openocd-build
cd openocd-0.10.0+dev && \
    ./bootstrap && \
sed -i 's/'"'"'openocd'"'"'/'"'"'openocd-mimiker'"'"'/g' configure
/bin/sh: 1: cd: can't cd to openocd-0.10.0+dev
Makefile:13: polecenia dla obiektu 'configure-stamp' nie powiodły się
make: *** [configure-stamp] Błąd 2
```

File: toolchain/openocd-mimiker/Makefile (hunk 2 above)
Comment (barnasm) [reply toolchain/openocd-mimiker/Makefile]: Added intermediate step to properly generate stamp files.

File: toolchain/openocd-mimiker/Makefile (hunk 3 above)
Comment (barnasm) [reply toolchain/openocd-mimiker/Makefile]: Added `distclean` rule.

**Pull request merged**